### PR TITLE
Volume Safe Delete

### PR DESCRIPTION
**Pull request by jtopjian**
This commit ensures that a volume is detached from all instances
before it is deleted.

It also adds in an `attachment` exported parameter that shows details
of the volume's attachment(s).

### CODE REVIEW

**Comment by jtopjian** [comment]
Let me know if there's anything wrong with this. I finished it quite late last night.

**Comment by julienvey** [comment]
Acceptance tests are successful in my environment, and the code looks good

@jrperritt Do you think we can merge this one, since it's the last one the TODO list of https://github.com/hashicorp/terraform/pull/924

**Comment by jtopjian** [comment]
Regarding the TODO: I think the floating IP issue needs sorted out before considering it ready to merge into terraform.

Summary:

nova-network floating IP support isn't implemented, and that's easily fixable by resurrecting #14. However, neutron floating IPs are broke in a bad way, as described in #14. Coincidentally, #14 also fixes that issue, but I haven't been able to get a solid answer on when _not_ to use `os-floating-ips` for neutron. So to be safe, neutron floating ip support needs fixed, which is where #18 came from. And while fixing that, I just kept coming across different things that could/should be done better, but might have made a mess out of the whole thing.

tl;dr: At minimum, #14 needs added as a quick fix.

**Comment by jrperritt** [comment]
I'll add it to the ToDo list.
